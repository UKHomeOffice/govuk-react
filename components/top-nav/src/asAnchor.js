import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { WHITE, YELLOW } from 'govuk-colours';

const asAnchor = (AnchorType) => {
  const Anchor = props => (
    <AnchorType {...props}>{props.children}</AnchorType>
  );

  Anchor.propTypes = {
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
      .isRequired,
  };

  const StyledHoc = glamorous(Anchor)({
    color: WHITE, // TODO: active state, LIGHT_BLUE
    textDecoration: 'none',
    borderBottom: '1px solid transparent',
    fontWeight: 700,
    ':hover': {
      borderBottomColor: WHITE, // TODO: active state, LIGHT_BLUE
    },
    ':focus': {
      backgroundColor: YELLOW,
      outline: `3px solid ${YELLOW}`,
    },
    ':visited': {
      color: WHITE, // TODO: active state, LIGHT_BLUE
      borderBottomColor: WHITE, // TODO: active state, LIGHT_BLUE
    },
  });

  return StyledHoc;
};

export default asAnchor;
