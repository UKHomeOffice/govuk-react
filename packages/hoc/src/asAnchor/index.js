import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { BLUE, PURPLE, YELLOW } from 'govuk-colours';

const asAnchor = (AnchorType) => {
  const Anchor = props => (
    <AnchorType {...props}>{props.children}</AnchorType>
  );

  Anchor.propTypes = {
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
      .isRequired,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
  };

  Anchor.defaultProps = {
    onClick: undefined,
    disabled: undefined,
  };

  const StyledHoc = glamorous(Anchor)({
    color: BLUE,
    padding: '3px',
    margin: '-3px',
    outlineColor: 'transparent',
    display: 'inline-block',
    textDecoration: 'underline',
    ':focus': {
      backgroundColor: YELLOW,
      outline: `3px solid ${YELLOW}`,
    },
    ':visited': {
      color: PURPLE,
    },
  });

  return StyledHoc;
};

export default asAnchor;
