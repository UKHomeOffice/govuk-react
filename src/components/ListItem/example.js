import React from "react";
import ListItem from "./index";

export default () => <ListItem>example</ListItem>;
