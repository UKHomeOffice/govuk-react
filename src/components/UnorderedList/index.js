// https://govuk-static.herokuapp.com/component-guide/government_navigation

import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';

const GUnorderedList = glamorous.ul(
  {
    boxSizing: 'border-box',
    width: '100%',
  },
  ({ listStyleType }) => ({
    listStyleType: listStyleType || undefined,
    marginLeft: listStyleType === 'none' ? 0 : undefined,
    paddingLeft: listStyleType === 'none' ? 0 : undefined,
  }),
);

// listStyleType: normal HTML property values are used here
// e.g., listStyleType="circle", listStyleType="upper-alpha", listStyleType="none"
const UnorderedList = ({ children, listStyleType }) => (
  <GUnorderedList listStyleType={listStyleType}>{children}</GUnorderedList>
);

UnorderedList.defaultProps = {
  listStyleType: undefined,
};

UnorderedList.propTypes = {
  children: PropTypes.node.isRequired,
  listStyleType: PropTypes.string,
};

export default UnorderedList;
