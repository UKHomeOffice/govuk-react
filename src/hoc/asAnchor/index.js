import React from "react";
import PropTypes from "prop-types";
import glamorous from "glamorous";
import { BLUE, PURPLE, YELLOW } from "govuk-colours";

const AnchorWrapper = glamorous.span({
  "> a": {
    color: BLUE,
    padding: "3px",
    margin: "-3px",
    outlineColor: "transparent",
    display: "inline-block",
    textDecoration: "underline",
    ":focus": {
      backgroundColor: YELLOW,
      outline: `3px solid ${YELLOW}`
    },
    ":visited": {
      color: PURPLE
    }
  }
});

const asAnchor = AnchorType => {
  const Anchor = props => (
    <AnchorWrapper>
      <AnchorType {...props}>{props.children}</AnchorType>
    </AnchorWrapper>
  );

  Anchor.propTypes = {
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
      .isRequired,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
  };

  Anchor.defaultProps = {
    onClick: undefined,
    disabled: undefined
  };

  return Anchor;
};

export default asAnchor;
