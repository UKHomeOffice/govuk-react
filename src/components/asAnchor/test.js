import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import asAnchor from "./";

const AnchorTag = asAnchor("a");

const wrapper = <AnchorTag href="https://example.com">Example</AnchorTag>;

describe(asAnchor, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches wrapper snapshot", () => {
    expect(shallow(wrapper)).toMatchSnapshot(`wrapper mount`);
  });
});
