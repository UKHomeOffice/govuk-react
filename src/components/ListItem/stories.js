import React from "react";
import { storiesOf } from "@storybook/react";

import ListItem from ".";

storiesOf("ListItem", module).add("ListItem", () => (
  <ListItem>ListItem example</ListItem>
));
