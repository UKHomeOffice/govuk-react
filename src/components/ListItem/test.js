import React from "react";
import ReactDOM from "react-dom";
import ListItem from "./";

describe(ListItem, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ListItem>Example</ListItem>, div);
  });
});
