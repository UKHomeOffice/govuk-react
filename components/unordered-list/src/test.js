import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import ListItem from '@govuk-react/list-item';
import asAnchor from '@govuk-react/hoc/lib/asAnchor';

import UnorderedList from './';

const AnchorTag = asAnchor('a');

const wrapper = (
  <UnorderedList>
    <ListItem><AnchorTag href="http://example.com">Example 1</AnchorTag></ListItem>
    <ListItem><AnchorTag href="http://example.com">Example 2</AnchorTag></ListItem>
    <ListItem><AnchorTag href="http://example.com">Example 3</AnchorTag></ListItem>
  </UnorderedList>
);

describe('UnorderedList', () => {
  it('renders with and without props, without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UnorderedList>Example</UnorderedList>, div);
    ReactDOM.render(
      <UnorderedList listStyleType="none">Example</UnorderedList>,
      div,
    );
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
