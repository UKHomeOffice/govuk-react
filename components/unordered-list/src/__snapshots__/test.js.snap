// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedComponent matches wrapper snapshot: wrapper mount 1`] = `
.emotion-19 {
  margin: 0;
  padding: 0;
  padding-left: 20px;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
}

@media only screen and (min-width:641px) {
  .emotion-19 {
    font-size: 16px;
    line-height: 1.25;
  }
}

.emotion-3 {
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
  width: 100%;
}

@media only screen and (min-width:641px) {
  .emotion-3 {
    font-size: 16px;
    line-height: 1.25;
    margin-bottom: 5px;
  }
}

.glamor-2,
[data-glamor-2] {
  margin-bottom: 0;
}

.glamor-0,
[data-glamor-0] {
  color: #005ea5;
  padding: 3px;
  margin: -3px;
  outline-color: transparent;
  display: inline-block;
  text-decoration: underline;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

.glamor-0:visited,
[data-glamor-0]:visited {
  color: #2e358b;
}

@media only screen and (min-width: 641px) {
  .glamor-2,
  [data-glamor-2] {
    margin-bottom: 0;
  }
}

<glamorous(UnorderedList)>
  <UnorderedList
    className="glamor-2"
  >
    <Styled(ul)
      className="glamor-2"
    >
      <ul
        className="glamor-2 glamor-19"
      >
        <glamorous(ListItem)>
          <ListItem
            className="glamor-2"
          >
            <Styled(li)
              className="glamor-2"
            >
              <li
                className="glamor-2 glamor-3"
              >
                <glamorous(Anchor)
                  href="http://example.com"
                >
                  <Anchor
                    className="glamor-0"
                    href="http://example.com"
                  >
                    <a
                      className="glamor-0"
                      href="http://example.com"
                    >
                      Example 1
                    </a>
                  </Anchor>
                </glamorous(Anchor)>
              </li>
            </Styled(li)>
          </ListItem>
        </glamorous(ListItem)>
        <glamorous(ListItem)>
          <ListItem
            className="glamor-2"
          >
            <Styled(li)
              className="glamor-2"
            >
              <li
                className="glamor-2 glamor-3"
              >
                <glamorous(Anchor)
                  href="http://example.com"
                >
                  <Anchor
                    className="glamor-0"
                    href="http://example.com"
                  >
                    <a
                      className="glamor-0"
                      href="http://example.com"
                    >
                      Example 2
                    </a>
                  </Anchor>
                </glamorous(Anchor)>
              </li>
            </Styled(li)>
          </ListItem>
        </glamorous(ListItem)>
        <glamorous(ListItem)>
          <ListItem
            className="glamor-2"
          >
            <Styled(li)
              className="glamor-2"
            >
              <li
                className="glamor-2 glamor-3"
              >
                <glamorous(Anchor)
                  href="http://example.com"
                >
                  <Anchor
                    className="glamor-0"
                    href="http://example.com"
                  >
                    <a
                      className="glamor-0"
                      href="http://example.com"
                    >
                      Example 3
                    </a>
                  </Anchor>
                </glamorous(Anchor)>
              </li>
            </Styled(li)>
          </ListItem>
        </glamorous(ListItem)>
      </ul>
    </Styled(ul)>
  </UnorderedList>
</glamorous(UnorderedList)>
`;
